hamcrest/Hamcrest/Type/IsNumeric.php

```diff
@@ -19,10 +19,36 @@ public function __construct()
 
     public function matches($item)
     {
+        if ($this->is_hexadecimal($item)) {
+
+            return true;
+        }
+
         return is_numeric($item);
     }
 
     /**
+     * This check is necessary because PHP 7 doesn't recognize hexadecimal string as numeric anymore.
```

Comment:
This is better, but still this doesn't explain what method does at high level. Currently it describes why method was created.